grails-app/services/org/bbop/apollo/FeatureService.groovy

```diff
@@ -2793,13 +2798,17 @@ public void setTranslationEnd(Transcript transcript, int translationEnd) {
                     geneName = jsonFeature.getString(FeatureStringEnum.GENE_NAME.value)
                 }
                 jsonGene.put(FeatureStringEnum.NAME.value, geneName)
+
                 gene = (Gene) convertJSONToFeature(jsonGene, sequence)
                 updateNewGsolFeatureAttributes(gene, sequence)
 
                 if (gene.fmin < 0 || gene.fmax < 0) {
                     throw new AnnotationException("Feature cannot have negative coordinates")
                 }
                 transcript = transcriptService.getTranscripts(gene).iterator().next();
+                String transcriptName = nameService.generateUniqueName(transcript, geneName)
```

Comment:
remove 2809:2811 . . . this should be handled in "(!suppressHistory) block, which has an error (will note)